lib/model/unreads.dart

```diff
@@ -353,7 +353,7 @@ class Unreads extends ChangeNotifier {
             } else {
               final messageIdsSet = Set.of(event.messages);
               mentions.removeAll(messageIdsSet);
-              _slowRemoveAllInStreams(messageIdsSet);
+              _removeAllInStreams(messageIdsSet);
```

Comment:
This version will make it slower if marking a long conversation as read — this loop is quadratic:
```dart
    for (var messageId in idsToRemove) {
      final info = _reverseStreamsLookup.remove(messageId);
      if (info != null && streams[info.streamId]?[info.topic] != null) {
        streams[info.streamId]![info.topic]!.remove(messageId);
```

because that `List.remove` call on a message in the middle of the list takes time linear in the number of remaining elements after it.

Probably the most efficient thing (in terms of our effort) would be to go ahead and add the new data structure, but leave the two "Enhance performance" commits aside and let #332 track the task of doing something like them.